pandas/io/parsers.py

```diff
@@ -348,38 +347,47 @@ def _validate_nrows(nrows):
     return nrows
 
 
+_compression_to_extension = {
```

Comment:
you can't sort None with strings in python 3.

```
 In [3]: set(['gz', 'xz']) | { None, 'infer'}
Out[3]: {None, 'xz', 'gz', 'infer'}

In [4]: sorted(set(['gz', 'xz']) | { None, 'infer'})
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-2dd21294aad8> in <module>()
----> 1 sorted(set(['gz', 'xz']) | { None, 'infer'})

TypeError: unorderable types: str() < NoneType()
```

instead you need to compare None separately (you can actually stringify things and it will work, e.g. None -> 'None') but that is non-pythonic